File: Labs/Lab 7/Lab 7.java
public int findTheCity(int n, int[][] edges, int distanceThreshold) {
        int[][] matrix = new int[n][n];
        for (int[] row : matrix) {
            Arrays.fill(row, Integer.MAX_VALUE/2);
        }

        for (int[] edge : edges) {
            int from = edge[0];
            int to = edge[1];
            int weight = edge[2];
            matrix[from][to] = weight;
            matrix[to][from] = weight;
        }

        for (int k = 0; k < n; k++) {
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    if (matrix[i][j] > matrix[i][k] + matrix[k][j]) {
                        matrix[i][j] = matrix[i][k] + matrix[k][j];
                    }
                }
            }
        }

        int minCount = n;
        int vertex = -1;

        
        for (int i = 0; i < n; i++) {
            int count = 0;
            for (int j = 0; j < n; j++) {
                if (i != j && matrix[i][j] <= distanceThreshold) {
                    count++;
                }
            }
            if (count <= minCount) {
                minCount = count;
                vertex = i;
            }
        }

        return vertex;
    }
}

File: Labs/Lab 7/Lab 7.py
def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        matrix = [[float('inf')] * n for _ in range(n)]

        for edge in edges:
            start, end, weight = edge
            matrix[start][end] = weight
            matrix[end][start] = weight

        for k in range(n):
            for i in range(n):
                for j in range(n):
                    matrix[i][j] = min(matrix[i][j], matrix[i][k] + matrix[k][j])

        minCount = n
        vertex = -1

        for i in range(n):
            count = sum(1 for j in range(n) if i != j and matrix[i][j] <= distanceThreshold)
            if count <= minCount:
                minCount = count
                vertex = i

        return vertex
